locations/pipelines/address_clean_up.py

```diff
@@ -16,8 +16,14 @@ def merge_address_lines(address_lines: [str]) -> str:
 def clean_address(address: list[str] | str) -> str:
     if not address:
         return ""
+
+    if isinstance(address, str):
+        if address.lower() == "undefined":
```

Comment:
Maybe include a step to remove whitespace?

```suggestion
        if address.strip().lower() == "undefined":
```